pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,92 +1,231 @@
-
-		err = survey.AskOne(&survey.Multiline{
-			Message: "oVirt certificate bundle",
-			Help:    fmt.Sprintf("The oVirt certificate bundle can be downloaded from %s.", pemURL),
-		},
-			&c.CABundle,
-			survey.ComposeValidators(survey.Required))
+
+		c, err = askPassword(c)
 		if err != nil {
-			return c, err
+			loginAttempts = loginAttempts - 1
+			logrus.Debugf("Login attempts now: %d", loginAttempts)
+			if loginAttempts == 0 {
+				return c, err
+			}
+		} else {
+			break
 		}
-	} else {
-		logrus.Warning("Communication with the oVirt engine will be insecure.")
 	}
+	return c, nil
+}
 
-	err = survey.Ask([]*survey.Question{
+// engineSetup will ask users: FQDN, execute validations and about
+// the credentials. In case of failure, returns Config and error
+func engineSetup() (Config, error) {
+	engineConfig := Config{}
+	httpResource := clientHTTP{}
+
+	err := survey.Ask([]*survey.Question{
 		{
 			Prompt: &survey.Input{
-				Message: "oVirt engine username",
-				Help:    "The user must have permissions to create VMs and disks on the Storage Domain with the same name as the OpenShift cluster.",
-				Default: "admin@internal",
+				Message: "Engine FQDN[:PORT]",
+				Help:    "The Engine FQDN[:PORT] (engine.example.com:443)",
 			},
 			Validate: survey.ComposeValidators(survey.Required),
 		},
-	}, &c.Username)
+	}, &engineConfig.FQDN)
 	if err != nil {
-		return c, err
+		return engineConfig, err
 	}
+	logrus.Debug("Engine FQDN: ", engineConfig.FQDN)
 
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Password{
-				Message: "oVirt engine password",
-				Help:    "",
-			},
-			Validate: survey.ComposeValidators(survey.Required, authenticated(&c)),
-		},
-	}, &c.Password)
+	// By default, we set Insecure true
+	engineConfig.Insecure = true
+
+	// Set c.URL with the API endpoint
+	engineConfig.URL = fmt.Sprintf("https://%s/ovirt-engine/api", engineConfig.FQDN)
+	logrus.Debug("Engine URL: ", engineConfig.URL)
+
+	// Start creating clientHTTP struct for checking if Engine FQDN is responding
+	httpResource.skipVerify = true
+	httpResource.urlAddr = engineConfig.URL
+	err = httpResource.checkURLResponse()
 	if err != nil {
-		return c, err
+		return engineConfig, err
 	}
 
-	return c, nil
+	// Set Engine PEM URL for Download
+	engineConfig.PemURL = fmt.Sprintf(
+		"https://%s/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA",
+		engineConfig.FQDN)
+	logrus.Debug("PEM URL: ", engineConfig.PemURL)
+
+	// Create tmpFile to store the Engine PEM file
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "engine-")
+	if err != nil {
+		logrus.Debug("Cannot create temporary file: ", err)
+		return engineConfig, err
+	}
+	defer os.Remove(tmpFile.Name())
+
+	// Download PEM
+	httpResource.saveFilePath = tmpFile.Name()
+	httpResource.skipVerify = true
+	httpResource.urlAddr = engineConfig.PemURL
+	err = httpResource.downloadFile()
+	if err != nil {
+		logrus.Warning("Cannot download PEM file from Engine!", err)
```

Comment:
Why do you keep going ahead to add a trust bundle that was never downloaded? Shouldn't we return here? Or something else